lib/tpm2_eventlog_yaml.c

```diff
@@ -384,6 +384,24 @@ static bool yaml_uefi_var(UEFI_VARIABLE_DATA *data, size_t size, UINT32 type,
                              uuidstr, sdata);
             free(sdata);
             return true;
+        } else if (type == EV_EFI_VARIABLE_BOOT) {
+            if ((strlen(ret) == 9 && strncmp(ret, "BootOrder", 9) == 0)) {
+                free(ret);
+                tpm2_tool_output("    VariableData:\n");
+                
+                if (data->VariableDataLength % 2 != 0) {
+                    LOG_ERR("BootOrder value length %" PRIu64 " is not divisible by 2\n",
+                            data->VariableDataLength);
+                    return false;
+                }
+
+                uint8_t *variable_data = (uint8_t *)&data->UnicodeName[
```

Comment:
```suggestion
                uint8_t *variable_data = &data->UnicodeName[
```
is the cast required?

Reply:
Right, the UnicodeName field is defined as `char16_t UnicodeName[];` Compiler didn't like it without an explicit cast, and would give the following error:
```
lib/tpm2_eventlog_yaml.c:398:42: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
                 uint8_t *variable_data = &data->UnicodeName[data->UnicodeNameLength];
```